resources/lib/functions.py

```diff
@@ -323,15 +323,18 @@ def getContent(url, params):
     if media_type.startswith("movie"):
         viewType = "Movies"
         xbmcplugin.setContent(pluginhandle, 'movies')
-    elif media_type == "music": # this is albums
+    elif media_type == "music":
+        viewType = "Artists"
```

Comment:
are you sure about this? music is a list of albums, how is the content artists?

Reply:
Yep, I'm sure. 
The Container.FolderPath of the artist node has "?media_type=music" in the URL

Reply:
that is weird, that is only set when type music is returned form the server

Reply:
we have to be very careful about what is content and what is the parent.
i.e. 
a parent item that is a set will produce content that is movies

Reply:
That's what is set in Kodi. 

Screenshot:
https://i.imgur.com/W7DNfI3.png?1

In this screen you also see that "4Lyn" has the "media_type=folder". The other ones have "media_type=MusicArtist".

Reply:
why are your albums of media type "Folder"?
there is something weird going on here

Reply:
That only applies for the artist "4Lyn" and 2 or 3 others ones.
All others are set correctly. Cannot tell you why.

Reply:
there is a weird thing going on here then, types are set by the item type, Folder is not mapped to anything or even valid.
also I dont know where this Folder Path come from, how did you get to this list, just by the addon menu? and selecting your music library?

Reply:
It's the addon node.

Reply:
how do you have your music organized, into albums or into artists folders?

Reply:
are you sure this is not from search results endpoint?

Reply:
I can not find anywhere in the addon that builds a addon path like the one in the "Folder Path" of your screen  shot.

Reply:
Artist/
   -- Album1
       --- Songs
   -- Album2
       --- Songs

Reply:
I may have to set this up to test.
My test is just 
album name
 - songs

Reply:
If you want I can give you access to my server, so you don't need to reproduce it on your side.

Reply:
this could be why you see music as the first name,  that is what is returned from the collection type and that is what is used as the first node type.
Because I have Album/Songs I see the content of the first node as albums
But you have Artist/Album/Songs so your first content is artists
I am not liking this.

Reply:
I guess.
I use the most common naming and folder structure:
https://github.com/MediaBrowser/Wiki/wiki/Music-naming

Reply:
I will need to set this up to test.

Reply:
ok, its all screwed, I need to go back to the drawing board on music

Reply:
Sorry ^^

Reply:
the root level is music as that is the collection type but after that the artist folders are considered just that Folders, not artists, the albums have correct types of MusicAlbum though so that is something.

Reply:
The problem is though if I start calling Folder = artist then that is going to cause issues if people are other structures, I would be locking the structure to this one.

Reply:
Music was and is always a pain. So many possible and mixed options to have them organized. Everybody use a different structure. It took my hours to get my music folders in shape when I moved over to the most common and my current structure.
Is there any reason why you have respect the folder structure? Doesn't Emby offer the option to gather the infos by artist/album/songs etc. via API?

Reply:
Yeah after sleeping on it I agree. I will change this so I control the structure of the returned data.

Reply:
Great. You should add a albums node listing too for people like you who prefer a sorting by albums.

Reply:
I dont a music collection, I have used spotify for years now.

Reply:
latest beta has a few changes for music and also the boxsets change from below